### PR TITLE
Hpu bf16

### PR DESCRIPTION
**Pull request by huijuanzh**
Enable bert,albert,debert,resnet50,swin-large on HPU BF16

### CODE REVIEW

File: byte_mlperf/backends/HPU/calculate_hpu_diff.py
Comment (JianzheXiao) [inline byte_mlperf/backends/HPU/calculate_hpu_diff.py:1]: do we really need this file? It looks like same with the file in cpu backend. We could use our original one directly

**Comment by ZhaiFeiyue** [comment]
@huijuanzh could you please also add  HPU in [here](https://github.com/bytedance/ByteMLPerf/blob/main/README.md?plain=1#L104C1-L104C1) and [here](https://github.com/bytedance/ByteMLPerf/blob/main/README.zh_CN.md?plain=1#L104)

File: byte_mlperf/backends/HPU/calculate_hpu_diff.py
Comment (huijuanzh) [reply byte_mlperf/backends/HPU/calculate_hpu_diff.py:1]: These two files are not used in HPU. Delete them. already add the commit

**Comment by ZhaiFeiyue** [comment]
@JianzheXiao any more comments?

File: byte_mlperf/reports/HPU/albert-torch-fp32/result-fp32.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+    "Model": "ALBERT-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_SQUAD",
+    "Instance Count": 1,
+    "Device Count": 128,
+    "Graph Compile": {
+        "Compile Duration": 5e-05,
+        "Compile Precision": "FP32",
```
Comment (Minghui-BD) [inline byte_mlperf/reports/HPU/albert-torch-fp32/result-fp32.json:10]: 这里应该BF16？

File: byte_mlperf/reports/HPU/swin-large-torch-fp32/open_imagenet.npy
Comment (Minghui-BD) [inline byte_mlperf/reports/HPU/swin-large-torch-fp32/open_imagenet.npy:1]: 这个麻烦删掉吧？

Comment (Minghui-BD) [reply byte_mlperf/reports/HPU/swin-large-torch-fp32/open_imagenet.npy:1]: @ZhaiFeiyue 麻烦看看哈，reports里精度标注不对，npy文件麻烦删掉哦~

Comment (ZhaiFeiyue) [reply byte_mlperf/reports/HPU/swin-large-torch-fp32/open_imagenet.npy:1]: @huijuanzh

File: byte_mlperf/reports/HPU/bert-torch-fp32/result-fp32.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+    "Model": "BERT-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_SQUAD",
+    "Instance Count": 1,
+    "Device Count": 128,
+    "Graph Compile": {
+        "Compile Duration": 3e-05,
+        "Compile Precision": "FP32",
```
Comment (ZhaiFeiyue) [inline byte_mlperf/reports/HPU/bert-torch-fp32/result-fp32.json:10]:
```suggestion
        "Compile Precision": "BF16",
```

File: byte_mlperf/reports/HPU/deberta-torch-fp32/result-fp32.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
+    "Model": "DEBERTA-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_SQUAD",
+    "Instance Count": 1,
+    "Device Count": 128,
+    "Graph Compile": {
+        "Compile Duration": 3e-05,
+        "Compile Precision": "FP32",
```
Comment (ZhaiFeiyue) [inline byte_mlperf/reports/HPU/deberta-torch-fp32/result-fp32.json:10]:
```suggestion
        "Compile Precision": "BF16",
```

File: byte_mlperf/reports/HPU/swin-large-torch-fp32/result-fp32.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
+{
+    "Model": "SWIN-LARGE-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_IMAGENET",
+    "Instance Count": 1,
+    "Device Count": 128,
+    "Graph Compile": {
+        "Compile Duration": 7e-05,
+        "Compile Precision": "FP32",
```
Comment (ZhaiFeiyue) [inline byte_mlperf/reports/HPU/swin-large-torch-fp32/result-fp32.json:10]:
```suggestion
        "Compile Precision": "BF16",
```

File: byte_mlperf/backends/HPU/compile_backend_hpu.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    def _update_model_env(self):
+        if self.model_info["model"] in ("bert-torch-fp32", "albert-torch-fp32"):
+            os.environ['LOWER_LIST'] ='byte_mlperf/backends/HPU/bert/bf16.txt'
+            os.environ['FP32_LIST'] ='byte_mlperf/backends/HPU/bert/fp32.txt'
+
+    def compile(self, config, dataloader=None):
+        result = {
+            "model": config['model_info']['model'],
+            "framework": config['model_info']['framework'],
+            "compile_precision": config['model_info']['model_precision'],
```
Comment (ZhaiFeiyue) [inline byte_mlperf/backends/HPU/compile_backend_hpu.py:45]:
```suggestion
            "compile_precision": “BF16”,
```

File: byte_mlperf/reports/HPU/albert-torch-fp32/result-fp32.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
+{
+    "Model": "ALBERT-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_SQUAD",
+    "Instance Count": 1,
+    "Device Count": 128,
```
Comment (Minghui-BD) [inline byte_mlperf/reports/HPU/albert-torch-fp32/result-fp32.json:7]: Device Count是卡数，不是cpu 核数

File: byte_mlperf/backends/HPU/compile_backend_hpu.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            os.environ['FP32_LIST'] ='byte_mlperf/backends/HPU/bert/fp32.txt'
+
+    def compile(self, config, dataloader=None):
+        result = {
+            "model": config['model_info']['model'],
+            "framework": config['model_info']['framework'],
+            "compile_precision": “BF16”,
+            "optimizations":{},
+            "instance_count": 1,
+            "device_count": 128,
```
Comment (ZhaiFeiyue) [inline byte_mlperf/backends/HPU/compile_backend_hpu.py:48]:
```suggestion
            "device_count": 1,
```

File: byte_mlperf/reports/HPU/albert-torch-fp32/result-fp32.json (hunk 6 above)
Comment (ZhaiFeiyue) [inline byte_mlperf/reports/HPU/albert-torch-fp32/result-fp32.json:7]:
```suggestion
    "Device Count": 1,
```

File: byte_mlperf/reports/HPU/bert-torch-fp32/result-fp32.json [hunk 8]
```diff
@@ -0,0 +1,57 @@
+{
+    "Model": "BERT-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_SQUAD",
+    "Instance Count": 1,
+    "Device Count": 128,
```
Comment (ZhaiFeiyue) [inline byte_mlperf/reports/HPU/bert-torch-fp32/result-fp32.json:7]:
```suggestion
    "Device Count": 1,
```

File: byte_mlperf/reports/HPU/deberta-torch-fp32/result-fp32.json [hunk 9]
```diff
@@ -0,0 +1,57 @@
+{
+    "Model": "DEBERTA-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_SQUAD",
+    "Instance Count": 1,
+    "Device Count": 128,
```
Comment (ZhaiFeiyue) [inline byte_mlperf/reports/HPU/deberta-torch-fp32/result-fp32.json:7]:
```suggestion
    "Device Count": 1,
```

File: byte_mlperf/reports/HPU/resnet50-torch-fp32/result-fp32.json [hunk 10]
```diff
@@ -0,0 +1,56 @@
+{
+    "Model": "RESNET50-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_IMAGENET",
+    "Instance Count": 1,
+    "Device Count": 128,
```
Comment (ZhaiFeiyue) [inline byte_mlperf/reports/HPU/resnet50-torch-fp32/result-fp32.json:7]:
```suggestion
    "Device Count": 1,
```

File: byte_mlperf/reports/HPU/swin-large-torch-fp32/result-fp32.json [hunk 11]
```diff
@@ -0,0 +1,56 @@
+{
+    "Model": "SWIN-LARGE-TORCH-FP32",
+    "Backend": "HPU",
+    "Host Info": "Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz",
+    "Dataset": "OPEN_IMAGENET",
+    "Instance Count": 1,
+    "Device Count": 128,
```
Comment (ZhaiFeiyue) [inline byte_mlperf/reports/HPU/swin-large-torch-fp32/result-fp32.json:7]:
```suggestion
    "Device Count": 1,
```

**Pull request merged**